packages/react-devtools-shared/src/backend/views/TraceUpdates/canvas.js

```diff
@@ -40,17 +52,21 @@ export function draw(nodeToData: Map<NativeType, Data>): void {
 
   const context = canvasFlow.getContext('2d');
   context.clearRect(0, 0, canvasFlow.width, canvasFlow.height);
-
-  nodeToData.forEach(({count, rect}) => {
+  iterateNodes(nodeToData, (rect, color) => {
     if (rect !== null) {
-      const colorIndex = Math.min(COLORS.length - 1, count - 1);
-      const color = COLORS[colorIndex];
-
-      drawBorder(context, rect, color);
+      drawBorder(context, rect, color)
     }
   });
 }
 
+function iterateNodes(nodeToData: Map<NativeType, Data>, execute: (rect: Rect, color: String) => void, node: NativeType) {
+  nodeToData.forEach(({count, rect}, node) => {
```

Comment:
Before, the code filters `rect !== null`. I'm not sure what it does, should we filter here for the native case?

Reply:
rect can be null when [`node.getBoundingClientRect` is not a function](https://github.com/facebook/react/blob/main/packages/react-devtools-shared/src/backend/views/TraceUpdates/index.js#L138). This is not right for RN as we don't have such method. Instead of do the feature detection in `index.js` file, I think it's better to leave that to the delegated canvas (draw method) in RN.